packages/chatbot-server-mongodb-public/src/processors/formatUserMessageForGeneration.ts

```diff
@@ -0,0 +1,48 @@
+/**
+  Adds front matter to the user message text.
+ */
+import { ConversationCustomData, logger } from "mongodb-rag-core";
+
+export function formatUserMessageForGeneration(
+  userMessageText: string,
+  customData?: ConversationCustomData
+) {
+  const frontMatters: Record<string, string> = {};
+  const { origin, originCode } = customData ?? {};
+  if (typeof origin !== "string" || !origin) {
+    return userMessageText;
+  }
+  try {
+    const url = new URL(origin);
+    if (
+      url.hostname === "mongodb.com" ||
+      url.hostname.endsWith(".mongodb.com")
+    ) {
+      frontMatters.pageUrl = origin;
+    }
+  } catch (e) {
+    logger.warn(`Origin ${origin} malformed. Not using in front matter.`);
+  }
+
+  if (originCode === "VSCODE") {
+    frontMatters.client = "MongoDB VS Code plugin";
+  } else if (originCode === "GEMINI_CODE_ASSIST") {
+    frontMatters.client = "Gemini Code Assist";
+  }
```

Comment:
rather than hard coding these values here, to be more maintainable, can you add them to the data structure of the origin codes wherever thats defined. 

also, perhaps we should just have a client label for all origin codes, e.g. 'MongoDB Docs website', "MongoDB University", etc.